components/classes/StudentsListPaper.tsx

```diff
@@ -1,264 +1,100 @@
-                </>
-              )}
-
-              {/* Nickname */}
-              {options.columns.includes("nickname") && (
-                <td>
-                  {student.nickname &&
-                    getLocaleString(student.nickname, options.language)}
-                </td>
-              )}
-
-              {options.columns.includes("allergies") && (
-                <td>{student.allergies?.join()}</td>
-              )}
-
-              {/* Shirt size */}
-              {options.columns.includes("shirtSize") && (
-                <td className="text-center">{student.shirt_size}</td>
-              )}
-
-              {/* Pants size */}
-              {options.columns.includes("pantsSize") && (
-                <td className="text-center">
-                  {student.pants_size?.replace("x", "×")}
-                </td>
-              )}
-
-              {/* Empty columns */}
-              {list(Math.min(options.numEmpty, MAXIMUM_EMPTY_COLUMNS) - 1).map(
-                (idx) => (
-                  <td
-                    key={idx}
-                    className={idx === 0 ? "!border-l-2" : undefined}
-                  >
-                    {idx === 0 && options.columns.length === 0
-                      ? " "
-                      : undefined}
-                  </td>
-                ),
-              )}
-
-              {/* Notes */}
-              {options.enableNotes && (
-                <td>{options.columns.length === 0 ? " " : undefined}</td>
-              )}
-            </tr>
-          ))}
-        </tbody>
-      </table>
-
-      {/* Timestamp */}
-      {options.enableTimestamp && (
-        <time className="mt-2 block text-sm opacity-50">
-          {new Date().toLocaleString(options.language, {
-            dateStyle: "full",
-            timeStyle: "medium",
-          })}
-        </time>
-      )}
-    </PaperPreview>
-  );
-};
+        {options.language === "en-US"
+          ? `M.${classroom.number}`
+          : `ม.${classroom.number}`}
+      </span>
+      <span className="mr-4 whitespace-nowrap font-bold">
+        {options.language === "en-US" ? "Class advisors" : "ครูที่ปรึกษา"}
+      </span>
+      <div className="mb-1 flex grow flex-row flex-wrap gap-x-2 font-bold">
+        {classroom.class_advisors.map((teacher) => (
+          <span key={teacher.id} className="-mb-1">
+            {getLocaleName(options.language, teacher, { prefix: true })}
+          </span>
+        ))}
+      </div>
+    </div>
+
+    {/* Table */}
+    <table
+      className={cn(`w-full border-2 border-black [&_td]:whitespace-nowrap
+        [&_td]:border-1 [&_td]:border-black [&_td]:px-1 [&_td]:py-0.5
+        [&_th]:border-1 [&_th]:border-black [&_th]:px-1 [&_th]:py-0.5`)}
+    >
+      {/* Head area */}
+      <StudentListTableHead
+        options={options}
+        className="border-b-2 border-black"
+      />
+
+      {/* Body area */}
+      <StudentListTableBody studentList={studentList} options={options} />

```

Comment:
Makes it easier to add new columns to the table because the table was seperated into 2 parts.